internal/preflight/preflight.go

```diff
@@ -77,12 +77,12 @@ func (p *preflight) PreflightUpgradeCheck(ctx context.Context, pv *kmmv1beta1.Pr
 		log.Info(utils.WarnString("failed to update the stage of Module CR in preflight to build stage"), "module", mod.Name, "error", err)
 	}
 
-	return p.helper.verifyBuild(ctx, mapping, mod, kernelVersion)
+	return p.helper.verifyBuild(ctx, pv, mapping, mod, kernelVersion)
 }
 
 type preflightHelperAPI interface {
 	verifyImage(ctx context.Context, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
-	verifyBuild(ctx context.Context, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
+	verifyBuild(ctx context.Context, pv *kmmv1beta1.PreflightValidation, mapping *kmmv1beta1.KernelMapping, mod *kmmv1beta1.Module, kernelVersion string) (bool, string)
```

Comment:
I am guessing that the `kernelVersion` inside `pv` is the new kernel version and the one passed stand alone is the current kernel version?

Can we rename those? I find it confusing and seems that we are just passing `kernelVersion` twice to function in different places in the code.

Reply:
actually they are the same. We do have duplicate data here. Since we need the PreflightValidation as part of the API, i will start using (passing) it into the functions, and will use its data inside functions as needed